pandas/core/series.py

```diff
@@ -378,103 +377,99 @@ def __init__(
                     data = SingleArrayManager.from_array(data, index)
             if copy:
                 data = data.copy()
-            if index is None:
-                index = data.index
-
-        else:
+            # skips validation of the name
+            object.__setattr__(self, "_name", name)
+            NDFrame.__init__(self, data)
+            return
```

Comment:
fastpath now ends here and everything below can simply be moved one level to the left.